### PR TITLE
NXP S32ZE update to HAL drivers version 2.0.0

### PR DESCRIPTION
**Pull request by haduongquang**
This PR updates S32ZE drivers to work on the Real-Time Drivers 2.0.0

### CODE REVIEW

File: drivers/ethernet/eth_nxp_s32_netc_psi.c [hunk 1]
```diff
@@ -259,7 +259,7 @@ BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(nxp_s32_netc_psi) == 1, "Only one PSI enabl
 #define INIT_VSIS(n)	DT_INST_NODE_HAS_PROP(n, vsis)
 
 #define NETC_PSI_INSTANCE_DEFINE(n)							\
-void nxp_s32_eth_psi##n##_rx_event(uint8_t chan, const uint32_t *buf, uint8_t buf_size)	\
+void nxp_s32_eth_psi##n##_rx_event(uint8_t chan, const uint32 *buf, uint8_t buf_size)	\
```
Comment (manuargue) [inline drivers/ethernet/eth_nxp_s32_netc_psi.c:262]:
```suggestion
void nxp_s32_eth_psi##n##_rx_event(uint8_t chan, const uint32_t *buf, uint8_t buf_size)	\
```

File: drivers/ethernet/eth_nxp_s32_netc_priv.h [hunk 2]
```diff
@@ -99,7 +99,7 @@
 #define _CONCAT3(a, b, c)	DT_CAT3(a, b, c)
 
 struct nxp_s32_eth_msix {
-	void (*handler)(uint8_t chan, const uint32_t *buf, uint8_t buf_size);
+	void (*handler)(uint8_t chan, const uint32 *buf, uint8_t buf_size);
```
Comment (manuargue) [inline drivers/ethernet/eth_nxp_s32_netc_priv.h:102]:
```suggestion
	void (*handler)(uint8_t chan, const uint32_t *buf, uint8_t buf_size);
```

File: drivers/ethernet/eth_nxp_s32_netc_psi.c [hunk 3]
```diff
@@ -359,27 +359,31 @@ static const Netc_Eth_Ip_EnetcGeneralConfigType nxp_s32_eth##n##_enetc_general_c
...
+	.maskMACVLANPromiscuousEnable =	(uint16)0x3U,					\
+	.maskVLANAllowUntaggedEnable =	(uint32)0x30000U,				\
 };											\
 											\
 static const Netc_Eth_Ip_StationInterfaceConfigType nxp_s32_eth##n##_si_cfg = {		\
 	.NumberOfRxBDR = 1,								\
 	.NumberOfTxBDR = 2,								\
 	.txMruMailboxAddr = NULL,							\
-	.rxMruMailboxAddr = (uint32_t *)MRU_MBOX_ADDR(DT_DRV_INST(n), rx),		\
+	.rxMruMailboxAddr = (uint32 *)MRU_MBOX_ADDR(DT_DRV_INST(n), rx),		\
```
Comment (manuargue) [inline drivers/ethernet/eth_nxp_s32_netc_psi.c:370]:
```suggestion
	.rxMruMailboxAddr = (uint32_t *)MRU_MBOX_ADDR(DT_DRV_INST(n), rx),		\
```

File: drivers/ethernet/eth_nxp_s32_netc_psi.c [hunk 4]
```diff
@@ -359,27 +359,31 @@ static const Netc_Eth_Ip_EnetcGeneralConfigType nxp_s32_eth##n##_enetc_general_c
...
 	.siMsgMruMailboxAddr = COND_CODE_1(INIT_VSIS(n),				\
-		((uint32_t *)MRU_MBOX_ADDR(DT_DRV_INST(n), vsi_msg)), (NULL)),		\
+		((uint32 *)MRU_MBOX_ADDR(DT_DRV_INST(n), vsi_msg)), (NULL)),		\
```
Comment (manuargue) [inline drivers/ethernet/eth_nxp_s32_netc_psi.c:372]:
```suggestion
		((uint32_t *)MRU_MBOX_ADDR(DT_DRV_INST(n), vsi_msg)), (NULL)),		\
```

File: drivers/ethernet/eth_nxp_s32_netc_vsi.c [hunk 5]
```diff
@@ -82,7 +82,7 @@ BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(nxp_s32_netc_vsi) == 1, "Only one VSI enabl
 #define NETC_VSI_INSTANCE_DEFINE(n)								\
 	NETC_GENERATE_MAC_ADDRESS(n)								\
 												\
-	void nxp_s32_eth_vsi##n##_rx_event(uint8_t chan, const uint32_t *buf, uint8_t buf_size)	\
+	void nxp_s32_eth_vsi##n##_rx_event(uint8_t chan, const uint32 *buf, uint8_t buf_size)	\
```
Comment (manuargue) [inline drivers/ethernet/eth_nxp_s32_netc_vsi.c:85]:
```suggestion
	void nxp_s32_eth_vsi##n##_rx_event(uint8_t chan, const uint32_t *buf, uint8_t buf_size)	\
```

File: drivers/ethernet/eth_nxp_s32_netc_vsi.c [hunk 6]
```diff
@@ -137,7 +137,8 @@ BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(nxp_s32_netc_vsi) == 1, "Only one VSI enabl
 		.NumberOfRxBDR = 1,								\
 		.NumberOfTxBDR = 1,								\
 		.txMruMailboxAddr = NULL,							\
-		.rxMruMailboxAddr = (uint32_t *)MRU_MBOX_ADDR(DT_DRV_INST(n), rx),		\
+		.rxMruMailboxAddr = (uint32 *)MRU_MBOX_ADDR(DT_DRV_INST(n), rx),		\
```
Comment (manuargue) [inline drivers/ethernet/eth_nxp_s32_netc_vsi.c:140]:
```suggestion
		.rxMruMailboxAddr = (uint32_t *)MRU_MBOX_ADDR(DT_DRV_INST(n), rx),		\
```

File: drivers/mbox/mbox_nxp_s32_mru.c [hunk 7]
```diff
@@ -80,9 +80,9 @@ static int nxp_s32_mru_send(const struct device *dev, uint32_t channel,
 	tx_cfg.NumTxMB = MRU_MAX_MBOX_PER_CHAN;
 	tx_cfg.LastTxMBIndex = MRU_MAX_MBOX_PER_CHAN - 1;
 	tx_cfg.MBAddList = (volatile uint32 * const *)tx_mbox_addr;
-	tx_cfg.ChMBSTATAdd = &cfg->base->CHXCONFIG[channel].CH_MBSTAT;
+	tx_cfg.ChMBSTATAdd = (volatile uint32 *)&cfg->base->CHXCONFIG[channel].CH_MBSTAT;
```
Comment (manuargue) [inline drivers/mbox/mbox_nxp_s32_mru.c:83]:
```suggestion
	tx_cfg.ChMBSTATAdd = (volatile uint32_t *)&cfg->base->CHXCONFIG[channel].CH_MBSTAT;
```

File: drivers/mbox/mbox_nxp_s32_mru.c [hunk 8]
```diff
@@ -80,9 +80,9 @@ static int nxp_s32_mru_send(const struct device *dev, uint32_t channel,
...
 
-	status = Mru_Ip_Transmit(&tx_cfg, (const uint32_t *)msg->data);
+	status = Mru_Ip_Transmit(&tx_cfg, (const uint32 *)msg->data);
```
Comment (manuargue) [inline drivers/mbox/mbox_nxp_s32_mru.c:85]:
```suggestion
	status = Mru_Ip_Transmit(&tx_cfg, (const uint32_t *)msg->data);
```

File: drivers/mbox/mbox_nxp_s32_mru.c [hunk 9]
```diff
@@ -208,11 +208,11 @@ static const struct mbox_driver_api nxp_s32_mru_driver_api = {
 	}
 
 #define MRU_CH_RX_CFG(i, n)								\
-	static volatile const uint32_t * const						\
+	static volatile const uint32 * const						\
```
Comment (manuargue) [inline drivers/mbox/mbox_nxp_s32_mru.c:211]:
```suggestion
	static volatile const uint32_t * const						\
```

File: drivers/mbox/mbox_nxp_s32_mru.c [hunk 10]
```diff
@@ -208,11 +208,11 @@ static const struct mbox_driver_api nxp_s32_mru_driver_api = {
...
 	nxp_s32_mru_##n##_ch_##i##_rx_mbox_addr[MRU_MAX_MBOX_PER_CHAN] = {		\
-		(uint32_t *const)MRU_MBOX_ADDR(n, i, 0),				\
+		(uint32 *const)MRU_MBOX_ADDR(n, i, 0),				\
```
Comment (manuargue) [inline drivers/mbox/mbox_nxp_s32_mru.c:213]:
```suggestion
		(uint32_t *const)MRU_MBOX_ADDR(n, i, 0),				\
```

File: drivers/mbox/mbox_nxp_s32_mru.c [hunk 11]
```diff
@@ -283,8 +283,8 @@ static const struct mbox_driver_api nxp_s32_mru_driver_api = {
 			.ChannelCfg = COND_CODE_0(MRU_RX_CHANNELS(n),			\
 						  (NULL), (nxp_s32_mru_##n##_ch_cfg)),	\
 			.NOTIFYAdd = {							\
-				&MRU_BASE(n)->NOTIFY[0],				\
-				&MRU_BASE(n)->NOTIFY[1]					\
+				(const volatile uint32 *)&MRU_BASE(n)->NOTIFY[0],	\
+				(const volatile uint32 *)&MRU_BASE(n)->NOTIFY[1]	\
```
Comment (manuargue) [inline drivers/mbox/mbox_nxp_s32_mru.c:287]:
```suggestion
				(const volatile uint32_t *)&MRU_BASE(n)->NOTIFY[0],	\
				(const volatile uint32_t *)&MRU_BASE(n)->NOTIFY[1]	\
```

File: drivers/mbox/mbox_nxp_s32_mru.c [hunk 12]
```diff
@@ -235,18 +235,18 @@ static const struct mbox_driver_api nxp_s32_mru_driver_api = {
...
-		.ChMBSTATAdd = &MRU_BASE(n)->CHXCONFIG[i].CH_MBSTAT,			\
+		.ChMBSTATAdd = (volatile uint32 *)&MRU_BASE(n)->CHXCONFIG[i].CH_MBSTAT,	\
 		.NumMailbox = MRU_MAX_MBOX_PER_CHAN,					\
 		.MBLinkReceiveChCfg = nxp_s32_mru_##n##_ch_##i##_rx_link_cfg		\
 	}
 
 /* Callback wrapper to adapt MRU's baremetal driver callback to Zephyr's mbox driver callback */
 #define MRU_CALLBACK_WRAPPER_FUNC(n)								\
-	void nxp_s32_mru_##n##_cb(uint8_t channel, const uint32_t *buf, uint8_t mbox_count)	\
+	void nxp_s32_mru_##n##_cb(uint8_t channel, const uint32 *buf, uint8_t mbox_count)	\
```
Comment (manuargue) [inline drivers/mbox/mbox_nxp_s32_mru.c:249]:
```suggestion
	void nxp_s32_mru_##n##_cb(uint8_t channel, const uint32_t *buf, uint8_t mbox_count)	\
```

File: drivers/mbox/mbox_nxp_s32_mru.c [hunk 13]
```diff
@@ -235,18 +235,18 @@ static const struct mbox_driver_api nxp_s32_mru_driver_api = {
 
 #define MRU_CH_CFG(i, n)								\
 	{										\
-		.ChCFG0Add = &MRU_BASE(n)->CHXCONFIG[i].CH_CFG0,			\
+		.ChCFG0Add = (volatile uint32 *)&MRU_BASE(n)->CHXCONFIG[i].CH_CFG0,	\
```
Comment (manuargue) [inline drivers/mbox/mbox_nxp_s32_mru.c:238]:
```suggestion
		.ChCFG0Add = (volatile uint32_t *)&MRU_BASE(n)->CHXCONFIG[i].CH_CFG0,	\
```

File: drivers/mbox/mbox_nxp_s32_mru.c [hunk 14]
```diff
@@ -235,18 +235,18 @@ static const struct mbox_driver_api nxp_s32_mru_driver_api = {
...
 		.ChCFG0 = RTU_MRU_CH_CFG0_IE(0) | RTU_MRU_CH_CFG0_MBE0(0),		\
-		.ChCFG1Add = &MRU_BASE(n)->CHXCONFIG[i].CH_CFG1,			\
+		.ChCFG1Add = (volatile uint32 *)&MRU_BASE(n)->CHXCONFIG[i].CH_CFG1,	\
```
Comment (manuargue) [inline drivers/mbox/mbox_nxp_s32_mru.c:240]:
```suggestion
		.ChCFG1Add = (volatile uint32_t *)&MRU_BASE(n)->CHXCONFIG[i].CH_CFG1,	\
```

File: drivers/mbox/mbox_nxp_s32_mru.c [hunk 15]
```diff
@@ -235,18 +235,18 @@ static const struct mbox_driver_api nxp_s32_mru_driver_api = {
...
 		.ChCFG1 = RTU_MRU_CH_CFG1_MBIC0(MRU_INT_GROUP(DT_INST_IRQN(n))),	\
-		.ChMBSTATAdd = &MRU_BASE(n)->CHXCONFIG[i].CH_MBSTAT,			\
+		.ChMBSTATAdd = (volatile uint32 *)&MRU_BASE(n)->CHXCONFIG[i].CH_MBSTAT,	\
```
Comment (manuargue) [inline drivers/mbox/mbox_nxp_s32_mru.c:242]:
```suggestion
		.ChMBSTATAdd = (volatile uint32_t *)&MRU_BASE(n)->CHXCONFIG[i].CH_MBSTAT,	\
```

File: drivers/mbox/mbox_nxp_s32_mru.c [hunk 16]
```diff
@@ -208,11 +208,11 @@ static const struct mbox_driver_api nxp_s32_mru_driver_api = {
...
 
 #define MRU_CH_RX_CFG(i, n)								\
-	static volatile const uint32_t * const						\
+	static volatile const uint32 * const						\
 	nxp_s32_mru_##n##_ch_##i##_rx_mbox_addr[MRU_MAX_MBOX_PER_CHAN] = {		\
-		(uint32_t *const)MRU_MBOX_ADDR(n, i, 0),				\
+		(uint32 *const)MRU_MBOX_ADDR(n, i, 0),				\
 	};										\
-	static uint32_t nxp_s32_mru_##n##_ch_##i##_buf[MRU_MAX_MBOX_PER_CHAN];		\
+	static uint32 nxp_s32_mru_##n##_ch_##i##_buf[MRU_MAX_MBOX_PER_CHAN];		\
```
Comment (manuargue) [inline drivers/mbox/mbox_nxp_s32_mru.c:215]:
```suggestion
	static uint32_t nxp_s32_mru_##n##_ch_##i##_buf[MRU_MAX_MBOX_PER_CHAN];		\
```

**Review by manuargue — CHANGES_REQUESTED** [review]

**Comment by manuargue** [comment]
pls fix the failed builds for canxl https://github.com/zephyrproject-rtos/zephyr/pull/76879/checks?check_run_id=28579473752

File: drivers/ethernet/eth_nxp_s32_netc_vsi.c (hunk 6 above)
Comment (haduongquang) [reply drivers/ethernet/eth_nxp_s32_netc_vsi.c:140]: The RTD has **uint32** type, which is long unsigned int, but Zephyr SDK is using **uint32_t** (unsigned int), so it makes the incompatible type build warning.

RTD: **uint32** <-> long unsigned int
Zephyr SDK: **uint32_t** <-> unsigned int

So I cast to **uint32** to fix the build warning.

image

**Comment by haduongquang** [comment]
> pls fix the failed builds for canxl https://github.com/zephyrproject-rtos/zephyr/pull/76879/checks?check_run_id=28579473752

I fixed

File: drivers/ethernet/eth_nxp_s32_netc_vsi.c (hunk 6 above)
Comment (manuargue) [reply drivers/ethernet/eth_nxp_s32_netc_vsi.c:140]: I see. If keeping the type defined by the HAL, then all cast that are done in the log msgs are unnecessary (e.g. `LOG_DBG("Tx Warning (error 0x%x)", (uint32_t)u32SysStatus);`, just fix the format string to the appropriate type.

Comment (haduongquang) [reply drivers/ethernet/eth_nxp_s32_netc_vsi.c:140]: I have updated the type of `u32SysStatus` and `buffidx` in the callback function to uint32_t, so we don't need to update or cast anything anymore.

**Comment by manuargue** [comment]
ping @henrikbrixandersen for the CAN changes, if you want to have a look

**Review by henrikbrixandersen — DISMISSED** [review]
CAN changes look fine, thanks.

**Review by manuargue — DISMISSED** [review]

**Comment by manuargue** [comment]
@haduongquang pls rebase & update manifest

**Comment by haduongquang** [comment]
I rebased `main` and updated `west.yml`

**Review by Dat-NguyenDuy — APPROVED** [review]

**Review by manuargue — APPROVED** [review]

**Comment by haduongquang** [comment]
I corrected the author of the second commit.

**Pull request merged**